datahub-graphql-core/src/main/resources/search.graphql

```diff
@@ -65,9 +65,14 @@ input SearchInput {
   count: Int
 
   """
-  Facet filters to apply to search results
+  Facet filters to apply to search results. These will be 'AND'-ed together.
   """
   filters: [FacetFilterInput!]
+
+  """
+  Filters are by default 'AND'-ed together. This lets you provide a set of OR filters
+  """
+  orFilters: [FacetFilterInput!]
```

Comment:
Could we use a nested model instead of reusing FacetFilterInput?

basically:


```
type OrFilter {
    and: [FacetFilterInput!] 
}
```

This is a standard way to define filters, it's [Disjunctive Normal Form ](https://en.wikipedia.org/wiki/Disjunctive_normal_form)

Reply:
The reason i call this out is that once we commit this code it's going to become very nontrivial to change this, for the same reason its hard to change the existing semantics around AND filters..

Reply:
And unsurprisingly, this is how we chose to model the Filter.pdl model already, because we were looking ahead at this. 

It's shaped as 


{
    or: [
       {
             and: [ { field: field, value: value } ]
       }
    ]
}

Reply:
updated!